### PR TITLE
[CHORE] Feature/aries oob connection invitation for mines

### PR DESCRIPTION
**Pull request by Jsyro**
## Objective 

Add API integration for Dev Traction and a way to create Out-of-Band invitations (which uses the DIDExchange protocol). mulitple invitations may need to be created, but each mine should only have one active connection. 

UI found here: https://traction-tenant-ui-dev.apps.silver.devops.gov.bc.ca/tenant/settings
API found here: https://traction-tenant-proxy-dev.apps.silver.devops.gov.bc.ca/api/doc

I have created a wallet for the 'Chief Permitting Officer' and have credentials to share with the team. My local development calls out to that dev hosted instance, however I could provide instructions to run traction locally if desired.

Does anything need to be modified for the new environment variables to injected once deployed? how would you like to feature flag this for TEST/PROD, as those wallets have not been created. 

sidenote: I also resolved the warnings about duplicate polymorphic identities, camp, building, and staging area were all being saved/loaded as the same types. some data cleanup would be needed if there those types exist in prod data.

### CODE REVIEW

**Review by matbusby-fw — DISMISSED** [review]
Looks solid to me.  I see a couple of things that I'm guessing are set up for further development, but it's all nice and clean.

You can rename your post title from [MDS-????] ... to [CHORE] ... to get past that one failing check.  Api tests seem to be failing as well.  Didn't look at them yet, but sometimes they can be a bit flaky and re-running them can help.

File: services/core-api/app/api/verifiable_credentials/models/connection.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    #ARIES-RFC 0023 https://github.com/hyperledger/aries-rfcs/tree/main/features/0023-did-exchange
+
+    invitation_id = db.Column(db.String)
+    
+    def __repr__(self):
+        return '<MineVerifiableCredentialConnection mine_guid=%r, connection_state=%r>' % self.mine_guid, self.connection_state or "UNKNOWN"
+
+    @classmethod
+    def find_by_mine_guid(cls, mine_guid):
+        return cls.query.filter_by(mine_guid=mine_guid, deleted_ind=False).all()
```
Comment (taraepp) [inline services/core-api/app/api/verifiable_credentials/models/connection.py]: I'm seeing on a failing test that deleted_ind doesn't exist here, but I'm not sure that the table exists either. I think a migration will be needed to add it. `migrations/sql/V2023.06.29.14.34__add_mine_document_version_table.sql` should be a good example of the format.

File: migrations/sql/V2023.09.08.15.10__add_verifiable_credential_connection_table.sql.sql [hunk 2]
```diff
@@ -0,0 +1,15 @@
+
+CREATE TABLE mine_verifiable_credential_connection
```
Comment (taraepp) [inline migrations/sql/V2023.09.08.15.10__add_verifiable_credential_connection_table.sql.sql]: probably want to fix .sql.sql. Validation performed on these file names gets tripped pretty easily.

File: services/core-api/app/api/verifiable_credentials/models/connection.py (hunk 1 above)
Comment (Jsyro) [reply services/core-api/app/api/verifiable_credentials/models/connection.py]: Fixed that one. I wasn't thinking this needs soft-deletes, but i could implemented it with soft-delete in mind.

File: migrations/sql/V2023.09.08.15.10__add_verifiable_credential_connection_table.sql.sql (hunk 2 above)
Comment (Jsyro) [reply migrations/sql/V2023.09.08.15.10__add_verifiable_credential_connection_table.sql.sql]: FIXED. yes flyway can be fickle

File: services/core-api/app/api/verifiable_credentials/resources/verifiable_credential.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
+from flask_restplus import Resource
+
+from app.extensions import api
+
+from app.api.utils.resources_mixins import UserMixin
+from app.api.services.traction_service import TractionService
+
+class VerifiableCredentialResource(Resource, UserMixin):
+    @api.doc(description='', params={})
+    def get(self):
```
Comment (simensma-fresh) [inline services/core-api/app/api/verifiable_credentials/resources/verifiable_credential.py:10]: Are you able to add a feature flag to these endpoints?

I added a flag named `verifiable_credentials` to our feature flag service and enabled it for local development and in our dev/test environments.

How?
1. Register the flag in `feature_flag.py`
2. Use the `is_feature_enabled` check

**Review by simensma-fresh — CHANGES_REQUESTED** [review]

File: services/core-api/app/api/verifiable_credentials/resources/verifiable_credential.py (hunk 3 above)
Comment (Jsyro) [reply services/core-api/app/api/verifiable_credentials/resources/verifiable_credential.py:10]: I added it to the namespace registration. i figured that was better than each individual endpoint

Comment (simensma-fresh) [reply services/core-api/app/api/verifiable_credentials/resources/verifiable_credential.py:10]: Good call!

**Review by simensma-fresh — APPROVED** [review]

**Review by matbusby-fw — APPROVED** [review]

**Comment by Jsyro** [comment]
I do have the 'MERGE' button as I do not have write access to this repo. So someone else will need to actually merge this REPO. 

Is there someone brave enough to hit the button?

**Pull request merged**